kubernetes/annotations_and_labels.md

```diff
@@ -1,21 +1,33 @@
 # Kubernetes Annotations and Labels
 
-## Common Labels
+## Common Labels (for host cluster services)
 
 - `app` - value should contain the name of the application. Should be applied
-  to every Kubernetes resource associated with an application. This should be
-  also used as replicas selector. E.g. `app=kvm-operator`.
+  to every Kubernetes resource associated with an application. This should
+  also be used as replicas selector. E.g. `app=kvm-operator`.
 - `giantswarm.io/cluster` - value should contain guest cluster ID which this
   object is part of. E.g. `giantswarm.io/cluster=eggs2`.
-  operator managing the object. E.g. `giantswarm.io/managed-by=kvm-operator`.
 - `giantswarm.io/managed-by` - value should contain repository name of the
+  operator managing the object. E.g. `giantswarm.io/managed-by=kvm-operator`.
 - `giantswarm.io/organization` - value should contain guest cluster's
-organization ID as displayed in the front-end
-`giantswarm.io/organization=track-hunter`.
+  organization ID as displayed in the front-end
+  `giantswarm.io/organization=track-hunter`.
 - `giantswarm.io/randomkey` - value should contain type of the randomkey stored
   in secret. E.g. `giantswarm.io/randomkey=encryption` for guest API -> etcd
   encryption at rest.
 
+## Common Labels (for guest cluster services)
+
+- `app` - value should contain the name of the application. Should be applied
+  to every Kubernetes resource associated with an application. This should 
+  also be used as replicas selector. E.g. `app=nginx-ingress-controller`.
+  Exceptions can be made to accomodate for adherence to existing selectors
+  upstream.
+- `giantswarm.io/type` - value should be either `system` for core system
+  services (i.e. K8s components) or `managed` for Giant Swarm managed services
+  (i.e. networking, DNS, monitoring, ingress controller, etc.). Latter would be
+  usually managed by `chart-operator`.
```

Comment:
I don't get when to use `system` and `managed`. My understanding is we should add label to all manifests in k8scloudconfig and all manifests in charts we deploy by chart-operator. But how to distinguish which one is `system` or `managed`?

Reply:
Yeah, maybe we just do a single `managed`? other things might be too confusing

Reply:
In that case the value carries zero information. So maybe we can establish two thigns:

- Add `ManagedBy` parameter in k8scloudconfig and set `giantswarm.io/managed-by={{ .ManagedBy }}` label to all manifests in the generated cloud config.
- Add `managedBy` value in `values.yaml` in helm charts and set `giantswarm.io/managed-by={{ .Values.managedBy }}` label to all manifests in the chart.

That way we don't have to distinguish between guest/host labels. We still can select manifests by the label name regardless of the value. As a bonus value carries some potentially valuable information.

Reply:
What would be the content of `ManagedBy`?

I'm hesitant to mix host and guest cluster as the goal of the labels is different.

For example: In the future we are planning to add services that might not be managed, yet, for services that are still in `incubation` or whatever you want to call it. I want to use this label to communicate to the customer what is giant swarm and managed, and what is giant swarm and maybe not yet managed.

Reply:
Currently, I'm not planning on them touching it, but more for communicating the facts of management levels to them on a technical level, and I kind of want to be prepared for different levels of support/management we might have.

Reply:
My understanding is we want to easily identify services in guest clusters that will be managed services. As opposed to system services that we run in all guest clusters.

So node-exporter is an example of a system component we always run for our monitoring.
Prometheus could be an example of a future managed service we provide but customers would opt in.

IMO it is useful to label the system components now. The managed services we can add later as we roll them out. The only thing I'm not keen on is the label name `/type` is very generic. I'd prefer `giantswarm.io/component-type` or `/service-type`.

Reply:
+1 on `/type` being very generic.

Reply:
I like `/service-type` and we can see with time if and how we want to split services between things like `managed` and others like `system` maybe